documentation/develop/web-ext-command-reference.md

```diff
@@ -286,11 +287,18 @@ If no target is specified, the extension will run in `firefox-desktop`.
 
 Environment variable: `$WEB_EXT_TARGET`
 
-### \--chromium-binary
+#### \--args, --arg
+Additional CLI options passed to the Browser binary. Example:
+
+{% highlight javascript %}
+--arg "'--search=mozilla'" --arg "'--new-tab=https://duckduckgo.com'"
```

Comment:
@lfilho  I think that the double-quoting (using both `"` and `'`) here on the --arg values is currently needed to workaround this yargs related web-ext issue mozilla/web-ext#1846, is that right?

That should be fixed once we release web-ext 4.3.0 (fixed by updating yargs see mozilla/web-ext#1902)

I think that is reasonable that the example here does work in the version of web-ext currently released (4.2.0) as it is, but maybe we can remove the double quotes once web-ext 4.3.0 has been released.

Reply:
Double-quoting shouldn't be needed in either case. The quotes only matter for the shell.

@rpl Even with the yargs bug, it can be worked around by `--arg=\ --search=mozilla`, for example.

I would go with `--arg=--search=mozilla` instead of `--arg --search=mozilla`, for clarity.

Reply:
Thanks @Rob--W and @rpl! @lfilho, want to update the patch with Rob's last comment? Then we'll approve + merge. :)

Reply:
Sorry, @Rob--W, your suggestion doesn't work. It seems to think `--search` as an arg to web-ext itself instead of passing it to the browser:

```
> web-ext run --args=--search=mozilla --browser-console --url ./__tests__/shared/test_page.html

web-ext run

Run the extension

Options:
  --version                        Show version number                 [boolean]
[...]
[...]

Unknown argument: search
```

Reply:
It's `--args=\ --search=mozilla` with `=`. But I only showed it to show that quotes aren't needed.

The second issue, `--args=--search=mozilla` doesn't work on the current version (4.2.0), but will work with the upcoming release, `web-ext` 4.3.0. We intend to release this version this week.

Reply:
Personally I think that the `--args=\ --search=mozilla` workaround that Rob did suggest risks to be even more confusing that the double-quotes.

@lfilho @Rob--W I briefly discussed about this PR with @caitmuenster today and agreed to proceed with the following steps:

- wait for the web-ext v4.3.0 release to get on npm (I just tagged the new v.4.3.0 release and the travis job should release it on npm as soon as all travis tasks did pass)
- then let @lfilho to try and confirm that mozilla/web-ext#1846 is fixed in web-ext v4.3.0 as expected (and that an example using `--args="--search=mozilla"` would work as expected)
- then update this PR to not using the double-quotes (nor the `--args=\ --search=mozilla` workaround that Rob did suggest, personally I think that it risks to be even more confusing that the double-quotes) and merge it on master

@lfilho @Rob--W @caitmuenster how that sounds?

Reply:
Sounds good to me. I'll do it over the weekend.

Reply:
Thanks so much @lfilho! FYI I will be out on PTO from June 25 - July 6. If your patch doesn't get reviewed and merged by someone else during that time I will do it when I get back. :)

Reply:
The below did work:

- ./node_modules/web-ext/bin/web-ext run --verbose --args="--search=mozilla"
- ./node_modules/web-ext/bin/web-ext run --verbose --args=--search=mozilla
- ./node_modules/web-ext/bin/web-ext run --verbose --args=--search="mozilla firefox" 
- ./node_modules/web-ext/bin/web-ext run --verbose --args="--search=mozilla firefox" (this is weird, should it?)

The below did NOT work:
- ./node_modules/web-ext/bin/web-ext run --verbose --args "--search=mozilla firefox" (based on your comments above this is expected, but most of the other args in this documentation use this format, so it's weird that for `args` we **must** use `=` version.

Reply:
Just published a new commit as agreed :)